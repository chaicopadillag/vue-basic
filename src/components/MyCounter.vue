<template>
  <section>
    <h2>Counter Setup</h2>
    <p>{{ count }}</p>
    <hr />
    <h2>Square</h2>
    <p>{{ square }}</p>
    <div class="flex gap-4">
      <button class="btn" type="button" @click="count++">Increment</button>
      <button class="btn" type="button" @click="count--">Decrement</button>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { computed, ref } from 'vue';

type CountType = {
  initialCount: number;
};

const { initialCount } = defineProps<CountType>();

const count = ref(initialCount);
const square = computed(() => count.value * count.value);
</script>
