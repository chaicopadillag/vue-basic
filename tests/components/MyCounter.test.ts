import MyCounter from '@/components/MyCounter.vue';
import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';

describe('<MyCounter />', () => {
  it('should render', () => {
    const wrapper = mount(MyCounter, {
      props: {
        initialCount: 10,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should contain title', () => {
    const initialCount = 10;
    const wrapper = mount(MyCounter, {
      props: { initialCount },
    });

    const [counter, square] = wrapper.findAll('p');

    expect(counter.text()).contain(initialCount);
    expect(square.text()).contain(initialCount * initialCount);
  });

  it('should increment count', async () => {
    const initialCount = 10;
    const wrapper = mount(MyCounter, {
      props: {
        initialCount,
      },
    });

    const button = wrapper.find('button');
    await button.trigger('click');

    const [counter, square] = wrapper.findAll('p');

    expect(counter.text()).contain(initialCount + 1);
    expect(square.text()).contain((initialCount + 1) * (initialCount + 1));
  });

  it('should decrement count tow biz', async () => {
    const initialCount = 10;
    const wrapper = mount(MyCounter, {
      props: {
        initialCount,
      },
    });

    const button = wrapper.findAll('button')[1];
    await button.trigger('click');
    await button.trigger('click');

    const [counter, square] = wrapper.findAll('p');

    expect(counter.text()).contain(initialCount - 2);
    expect(square.text()).contain((initialCount - 2) * (initialCount - 2));
  });
});
