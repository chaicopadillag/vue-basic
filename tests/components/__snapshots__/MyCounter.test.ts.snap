// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter /> > should render 1`] = `
"<section>
  <h2>Counter Setup</h2>
  <p>10</p>
  <hr>
  <h2>Square</h2>
  <p>100</p>
  <div class="flex gap-4"><button class="btn" type="button">Increment</button><button class="btn" type="button">Decrement</button></div>
</section>"
`;
